src/components/ToggleDarkMode.tsx

```diff
@@ -8,8 +8,9 @@ const ToggleDarkMode: React.FC = () => {
   const [isDarkMode, setIsDarkMode] = useState(backgroundColor === "#121212");
 
   useEffect(() => {
-    setIsDarkMode(backgroundColor === "#121212");
-  }, [backgroundColor]);
+    const savedTheme = localStorage.getItem('theme');
```

Comment:
Can't we just get this before initialising the state variable above? We can get rid of this useState?